ui/src/adapters/api/credentials.ts

```diff
@@ -87,12 +161,91 @@ export const credentialStatusParser = getStrictParser<CredentialStatus>()(
 export async function getCredential({
   credentialID,
   env,
+  issuerIdentifier,
   signal,
 }: {
   credentialID: string;
   env: Env;
+  issuerIdentifier: IssuerIdentifier;
   signal?: AbortSignal;
 }): Promise<Response<Credential>> {
+  try {
+    const credentialDetail = await getCredentialDetail({
+      credentialID,
+      env,
+      issuerIdentifier,
+      signal,
+    });
+
+    if (!credentialDetail.success) {
+      return buildErrorResponse(credentialDetail.error);
+    }
+
+    const {
+      "@context": context,
+      credentialSchema,
+      credentialStatus,
+      credentialSubject,
+      expirationDate,
+      id,
+      issuanceDate,
+      issuer,
+      proof,
+    } = credentialDetail.data;
+
+    const revocationStatus = await getRevocationStatus({
+      env,
+      issuerIdentifier,
+      params: { nonce: credentialStatus.revocationNonce },
+      signal,
+    });
+
+    if (!revocationStatus.success) {
+      return buildErrorResponse(revocationStatus.error);
+    }
```

Comment:
Same as before, no need to rebuild an `ErrorResponse`. When `!revocationStatus.success`, `revocationStatus` is already an `ErrorResponse`.

```
    if (!revocationStatus.success) {
      return revocationStatus;
    }
```